lib/shrinkwrap.js

```diff
@@ -0,0 +1,118 @@
+const path = require('path');
+
+const _ = require('lodash');
+const traverse = require('traverse');
+
+const deps = require('./util/deps');
+const exec = require('./util/exec');
+const link = require('./util/link');
+const pfs = require('./util/pfs');
+const promise = require('./util/promise');
+
+module.exports = function(dir) {
+  var dependencyKey = this.dependencyKey;
+  var moduleDir = this.moduleDir;
+  var log = this.emit.bind(this);
+
+  var shrinkwrapCache = {};
+  var shrinkwrapNodeDeps = _.memoize(function(repo) {
+    var repoPath = repo.welder.repoPath;
+    var shrinkwrapPath = path.join(repoPath, 'npm-shrinkwrap.json');
+    var repoDeps = repo[dependencyKey];
+    // Compare names in welder dependencies against npm dependencies. If the
+    // dependency only exist in welder deps that unlink it so npm shrinkwrap
+    // won't error when it encounters the extraneous dependency.
+    var repoNpmDeps = repo.dependencies;
+    return promise.resolve(_.pairs(repoDeps))
```

Comment:
Feels like this should be extracted to a separate lib that does this portion of the job explicitly.